app/3.0/toolbox/case-study.md

```diff
@@ -107,38 +104,43 @@ When you change pages, the application loads the definition for the active view.
 When the definition loads, the browser _upgrades_ the element to a fully-
 functional custom element.
 
-```javascript
+`shop-app.js` { .caption }
+
+```js
 _pageChanged(page, oldPage) {
   if (page != null) {
-    // home route is eagerly loaded
-    if (page == 'home') {
-      this._pageLoaded(Boolean(oldPage));
-    // other routes are lazy loaded
-    } else {
-      // When a load failed, it triggered a 404 which means we need to
-      // eagerly load the 404 page definition
-      let cb = this._pageLoaded.bind(this, Boolean(oldPage));
-      Polymer.importHref(
-        this.resolveUrl('shop-' + page + '.html'),
-        cb, cb, true);
+    let cb = this._pageLoaded.bind(this, Boolean(oldPage));
+    switch (page) {
+      case 'list':
+        import('./shop-list.js').then(cb);
+        break;
+      case 'detail':
+        import('./shop-detail.js').then(cb);
+        break;
+      case 'cart':
+        import('./shop-cart.js').then(cb);
+        break;
+      case 'checkout':
+        import('./shop-checkout.js').then(cb);
+        break;
+      default:
+        this._pageLoaded(Boolean(oldPage));
     }
   }
 }
 ```
 
 In the logic above, the home view is built into the app shell, but the other
-views are demand-loaded fragments.
+views are [dynamic imports](https://developers.google.com/web/updates/2017/11/dynamic-import).
 
 Shop also uses [`dom-if`](/{{{polymer_version_dir}}}/docs/api/elements/Polymer.DomIf) templates to lazily create views:
 
 ```html
 <!-- Lazy-create the tabs for larger screen sizes. -->
-<div id="tabContainer" sticky$="[[_shouldShowTabs]]" hidden$="[[!_shouldShowTabs]]">
+<div id="tabContainer" sticky\$="[[_shouldShowTabs]]" hidden\$="[[!_shouldShowTabs]]">
```

Comment:
whats up with these `\$`? are they escaping?

Reply:
I have no idea what happened there, thanks for picking that up!

Reply:
Here it is in the source:

https://github.com/Polymer/shop/blob/c0017828188ad576f0dde5dbe17d1dd277b12b26/src/shop-app.js#L259

Reply:
I took out the line.. I think the sample is more focused on the dom-if anyway